src/main/kotlin/rx/lang/kotlin/observables.kt

```diff
@@ -2,71 +2,60 @@ package rx.lang.kotlin
-    override fun iterator(): Iterator<T> = this@toIterable
-}
-
-fun BooleanArray.toObservable() : Observable<Boolean> = this.toList().toObservable()
-fun ByteArray.toObservable() : Observable<Byte> = this.toList().toObservable()
-fun ShortArray.toObservable() : Observable<Short> = this.toList().toObservable()
-fun IntArray.toObservable() : Observable<Int> = this.toList().toObservable()
-fun LongArray.toObservable() : Observable<Long> = this.toList().toObservable()
-fun FloatArray.toObservable() : Observable<Float> = this.toList().toObservable()
-fun DoubleArray.toObservable() : Observable<Double> = this.toList().toObservable()
-fun <T> Array<out T>.toObservable() : Observable<T> = Observable.from(this)
-
-fun IntProgression.toObservable() : Observable<Int> =
+fun BooleanArray.toObservable(): Observable<Boolean> = this.toList().toObservable()
+fun ByteArray.toObservable(): Observable<Byte> = this.toList().toObservable()
+fun ShortArray.toObservable(): Observable<Short> = this.toList().toObservable()
+fun IntArray.toObservable(): Observable<Int> = this.toList().toObservable()
+fun LongArray.toObservable(): Observable<Long> = this.toList().toObservable()
+fun FloatArray.toObservable(): Observable<Float> = this.toList().toObservable()
+fun DoubleArray.toObservable(): Observable<Double> = this.toList().toObservable()
+fun <T> Array<out T>.toObservable(): Observable<T> = Observable.from(this)
+
+fun IntProgression.toObservable(): Observable<Int> =
         if (step == 1 && last.toLong() - first < Integer.MAX_VALUE) Observable.range(first, Math.max(0, last - first + 1))
         else Observable.from(this)
 
-fun <T> Iterator<T>.toObservable() : Observable<T> = toIterable().toObservable()
-fun <T> Iterable<T>.toObservable() : Observable<T> = Observable.from(this)
-fun <T> Sequence<T>.toObservable() : Observable<T> = Observable.from(object : Iterable<T> {
+fun <T> Iterator<T>.toObservable(): Observable<T> = toIterable().toObservable()
+fun <T> Iterable<T>.toObservable(): Observable<T> = Observable.from(this)
+fun <T> Sequence<T>.toObservable(): Observable<T> = Observable.from(object : Iterable<T> {
     override fun iterator(): Iterator<T> = this@toObservable.iterator()
 })
 
-fun <T> T.toSingletonObservable() : Observable<T> = Observable.just(this)
-fun <T> Throwable.toObservable() : Observable<T> = Observable.error(this)
+fun <T> Throwable.toObservable(): Observable<T> = Observable.error(this)
 
-fun <T> Iterable<Observable<out T>>.merge() : Observable<T> = Observable.merge(this.toObservable())
-fun <T> Iterable<Observable<out T>>.mergeDelayError() : Observable<T> = Observable.mergeDelayError(this.toObservable())
+fun <T> Iterable<Observable<out T>>.merge(): Observable<T> = Observable.merge(this.toObservable())
+fun <T> Iterable<Observable<out T>>.mergeDelayError(): Observable<T> = Observable.mergeDelayError(this.toObservable())
 
+fun <T, R> Observable<T>.fold(initial: R, body: (R, T) -> R): Observable<R> = reduce(initial, { a, e -> body(a, e) })
```

Comment:
If we create such aliases to align with Kotlin operators, here is no reason to alias `take()` to `drop()`, `takeWhile()` to `dropWhile()`, and `toMap()` to `associate()`. This seems fine if we truly want to push Kotlin's conventions, and ReactiveX has a few aliases of its own (e.g. `sample()` and `throttleLast()`. But my concern is that folks will see these aliases as legitimate ReactiveX operators. When they go [look up a marble diagram](http://rxmarbles.com/) for `drop()`, they will not find one. It can make the operators a bit convoluted documentation-wise. 

Plus `reduce()` and `fold()` mean two different things in Kotlin sequences and collections (one has a seed and the other doesn't) which adds to the confusion since ReactiveX has `reduce()` do both. My personal sentiment is we keep Kotlin and Rx operators distinctly separate. But if you think adopting the Kotlin naming conventions as aliases would be helpful, I won't resist that.